src/benchmark.rs

```diff
@@ -0,0 +1,107 @@
+use regex::Regex;
+use tabwriter::TabWriter;
+
+use std::io;
+use std::io::prelude::*;
+use std::io::BufReader;
+use std::fs::File;
+
+use utils::{drop_commas_and_parse, fmt_thousands_sep};
+
+/// All extractable data from a single micro-benchmark.
+#[derive(Debug)]
+pub struct Benchmark {
+    pub name: String,
+    pub ns: usize,
+    pub variance: usize,
+    pub throughput: Option<usize>,
+}
+
+/// A comparison between an old and a new benchmark.
+/// All differences are reported in terms of measuring improvements
+/// (negative) or regressions (positive). That is, if an old benchmark
+/// is slower than a new benchmark, then the difference is negative.
+/// Conversely, if an old benchmark is faster than a new benchmark,
+/// then the difference is positive.
+#[derive(Debug)]
+pub struct Comparison {
+    pub fst: Benchmark,
+    pub snd: Benchmark,
+    pub diff_ns: i64,
+    pub diff_ratio: f64,
+}
+
+const BENCHMARK_REGEX: &'static str = concat!(r#"test\s+(?P<name>\S+)\s+"#,
+                                              r#"... bench:\s+(?P<ns>[0-9,]+)\s+ns/iter"#,
+                                              r#"\s+\(\+/-\s+(?P<variance>[0-9,]+)\)"#,
+                                              r#"(?:\s+=\s+(?P<throughput>[0-9,]+))?"#);
```

Comment:
Maybe use `lazy_static!` here? (Or move the regex itself down into `parse`?) That way we don't have it defined across two places.

Also, it might be better to use `(?x)` to spread the regex over multiple lines, e.g.,

```
const BENCHMARK_REGEX: &'static str = r#"(?x)
test\s+(?P<name>\S+)\s+
...\sbench:\s+(?P<ns>[0-9,]+)\s+ns/iter
\s+\(\+/-\s+(?P<variance>[0-9,]+)\)
(?:\s+=\s+(?P<throughput>[0-9,]+))?
"#;
```